py/torch_tensorrt/dynamo/conversion/impl/matmul.py

```diff
@@ -48,6 +49,17 @@ def matrix_multiply(
     input, other = broadcast(
         ctx, input, other, f"{name}_input", f"{name}_other", preset_diff
     )
+
+    promoted_type = _enums.dtype._from(
+        torch.promote_types(
+            _enums.dtype._from(input.dtype).to(torch.dtype),
+            _enums.dtype._from(other.dtype).to(torch.dtype),
+        )
+    )
+    trt_promoted_type = promoted_type.to(trt.DataType)
+    input = cast_trt_tensor(ctx, input, trt_promoted_type, f"{name}_input_casted")
+    other = cast_trt_tensor(ctx, other, trt_promoted_type, f"{name}_other_casted")
+
```

Comment:
why are these casts needed ? Ideally, they should be already in the correct types? Did you encounter any failure cases ?

Reply:
Without this cast, SD UNet model failed with below error. This was caused strongly type network config.
torch_tensorrt [TensorRT Conversion Context][ERROR]ITensor::getDimensions: Error Code 4: API Usage Error ([MATRIX_MULTIPLY]-[torchtrt_lowered_ops.scaled_dot_product_attention]-[scaled_dot_product_attention_1_mm]: IMatrixMultiplyLayer must have same input types. `A` is of type Float and `B` is of type Half.)

I think I can add below condition
`if ctx.net.get_flag(trt.NetworkDefinitionCreationFlag.STRONGLY_TYPED)`